src/xdsl/dialects/irdl.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from xdsl.dialects.builtin import ArrayAttr, StringAttr
+
+from xdsl.irdl import *
+from xdsl.ir import *
+
+
+@dataclass
+class IRDL:
+    ctx: MLContext
+
+    def __post_init__(self):
+        self.ctx.register_attr(EqTypeConstraintAttr)
+        self.ctx.register_attr(AnyTypeConstraintAttr)
+        self.ctx.register_attr(AnyOfTypeConstraintAttr)
+        self.ctx.register_attr(VarTypeConstraintAttr)
+        self.ctx.register_attr(DynTypeBaseConstraintAttr)
+        self.ctx.register_attr(DynTypeParamsConstraintAttr)
+        self.ctx.register_attr(TypeParamsConstraintAttr)
+        self.ctx.register_attr(NamedTypeConstraintAttr)
+
+        self.ctx.register_op(DialectOp)
+        self.ctx.register_op(ParametersOp)
+        self.ctx.register_op(TypeOp)
+        self.ctx.register_op(ConstraintVarsOp)
+        self.ctx.register_op(OperandsOp)
+        self.ctx.register_op(ResultsOp)
+        self.ctx.register_op(OperationOp)
+
+
+@irdl_attr_definition
+class EqTypeConstraintAttr(ParametrizedAttribute):
+
+    name = "equality_type_constraint"
```

Comment:
Yeah, just realised that the attributes are given the same name but the operations are prefixed with the dialect. I'll make this quick change. 

Will this also apply to the operation in builtin? i see that it's missing the prefix ```builtin.```

Reply:
I know that MLIR is not entirely consistent here either (`llvm.struct` or `async.group` vs. `f32` or `i32`). IMO, the builtin types can be not prefaced because the builtin dialect is special either way considering that it provides things that everyone needs anyhow. But I don't have a strong opinion, maybe @math-fehr can indicate what he'd prefer?